src/google/protobuf/compiler/command_line_interface_unittest.cc

```diff
@@ -940,6 +959,53 @@ TEST_F(CommandLineInterfaceTest, WriteTransitiveDescriptorSetWithSourceInfo) {
   EXPECT_TRUE(descriptor_set.file(1).has_source_code_info());
 }
 
+TEST_F(CommandLineInterfaceTest, WriteDependencyManifestFile) {
+  CreateTempFile("foo.proto",
+    "syntax = \"proto2\";\n"
+    "message Foo {}\n");
+  CreateTempFile("bar.proto",
+    "syntax = \"proto2\";\n"
+    "import \"foo.proto\";\n"
+    "message Bar {\n"
+    "  optional Foo foo = 1;\n"
+    "}\n");
+
+  Run("protocol_compiler --dependency_out=$tmpdir/manifest "
+      "--test_out=$tmpdir --proto_path=$tmpdir bar.proto");
+
+  ExpectNoErrors();
+
+  ExpectFileContent("manifest",
+                    "$tmpdir/manifest: $tmpdir/foo.proto\\\n"
```

Comment:
normally the file would be:
<outputfile1> <outputfile2> ... : <dependency1> \\
  <dependency2>  \\
  <dependency3> \\
 ...

where for a main.c of:
#include "main.h"

int main() {
return 0;
}

cc -MMD main.c

you are going to get:

main.o: main.c main.h

our code is not going to set <outputfile> correctly. I would expect for the cpp compiler that we would actually generate 

foo.pb.cc foo.pb.h : foo.proto

Thoughts? I think the current code will actually work for my needs, but it is certainly not a correct solution as far as I can tell.

I'm working from "http://make.mad-scientist.net/papers/advanced-auto-dependency-generation/".